blivet/arch.py

```diff
@@ -198,6 +199,29 @@ def is_cell():
     return False
 
 
+def is_t2mac():
+    """
+    :return: True if the hardware is an Intel-based Apple Mac with the T2 security chip, False otherwise.
+    :rtype boolean
+
+    """
+    APPLE_T2_PRODUCT_NAMES = {
+        "iMac20,2", "iMac20,1", "iMacPro1,1",
+        "MacPro7,1", "Macmini8,1", "MacBookAir9,1", "MacBookAir8,2",
+        "MacBookAir8,1", "MacBookPro16,4", "MacBookPro16,3", "MacBookPro16,2",
+        "MacBookPro16,1", "MacBookPro15,4", "MacBookPro15,3", "MacBookPro15,2",
+        "MacBookPro15,1",
+    }
+    if not is_x86():
+        t2_mac = False
+    elif not os.path.isfile(DMI_PRODUCT_ID):
+        t2_mac = False
+    else:
+        buf = open(DMI_PRODUCT_ID).read()
```

Comment:
Leaking file descriptor.

Reply:
Ok, we do exactly the same below with the `DMI_CHASSIS_VENDOR`. I guess I can fix both later in a separate PR.

Reply:
Should I leave this as is.

Reply:
Yes, I will fix this later, we have this problem at more places. I want to give some more time to other people to review this, but other than that, this is ready to be merged. Thank you again.